src/resourcepacks/ResourcePack.php

```diff
@@ -45,6 +45,12 @@ public function getPackSize() : int;
 	 */
 	public function getPackVersion() : string;
 
+	/**
+	 * Returns the key with which the pack was encrypted.
+	 * @return string empty string if the pack is not encrypted
+	 */
+	public function getEncryptionKey() : string;
```

Comment:
This is an API breaking change.

Reply:
A better solution might be to add an EncryptedResourcePack interface that extends this one, or perhaps just keep the keys in ResourcePackManager directly.